lib/percy/network_helpers.rb

```diff
@@ -0,0 +1,53 @@
+require 'socket'
```

Comment:
This has been extracted 1-to-1, with minimal changes. Updated the method params in backwards compatible ways to expose a few more options like `retry_wait_seconds` which makes testing easier. Everything else in this PR is new.